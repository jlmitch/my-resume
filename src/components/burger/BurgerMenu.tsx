import React from 'react';
import { Link } from 'gatsby';
import { stack as Menu } from 'react-burger-menu';
import { burgerMenuStyles } from './burgerMenuStyles';

const BurgerMenu = () => (
  <Menu right styles={burgerMenuStyles}>
    <Link className={`nav-link ${location.pathname === '/' ? 'active' : '' }`} to="/">Home</Link>
    <Link className={`nav-link ${location.pathname === '/about/' ? 'active' : '' }`} to="/about">About</Link>
    <Link className={`nav-link ${location.pathname === '/resume/' ? 'active' : '' }`} to="/resume">Resume</Link>
    <Link className={`nav-link ${location.pathname === '/blog/' ? 'active' : '' }`} to="/blog">Blog</Link>
  </Menu>
);

export default BurgerMenu;
