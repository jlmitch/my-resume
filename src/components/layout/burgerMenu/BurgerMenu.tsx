import React, { useState } from 'react';
import styled from 'styled-components';
import MenuTray from './MenuTray';

interface StyledBurgerProps {
  open: boolean;
  onClick: () => void;
}

const StyledBurger = styled.div<StyledBurgerProps>`
  width: 2rem;
  height: 3rem;
  position: fixed;
  top: 25px;
  right: 40px;
  z-index: 20;
  display: flex;
  justify-content: space-around;
  flex-flow: column nowrap;

  /* Small devices such as large phones (640px and up) */
  @media only screen and (min-width: 40em) {
    display: none;
  }
  
  div {
    width: 3rem;
    height: 0.25rem;
    background-color: ${({ open }) => open ? '#0A1828' : '#178582'};
    border-radius: 10px;
    transform-origin: 1px;
    transition: all 0.3s linear;

    &:nth-child(1) {
      transform: ${({ open }) => open ? 'rotate(45deg)' : 'rotate(0)'};
    }

    &:nth-child(2) {
      transform: ${({ open }) => open ? 'translateX(100%)' : 'translateX(0)'};
      opacity: ${({ open }) => open ? 0 : 1};
    }

    &:nth-child(3) {
      transform: ${({ open }) => open ? 'rotate(-45deg)' : 'rotate(0)'};
    }
  }
`;
interface BurgerMenuProps {
  pathname: string;
}

const BurgerMenu = ({ pathname }: BurgerMenuProps) => {
  const [open, setOpen] = useState(false);

  const handleOnclick = () => setOpen(!open);

  return (
    <>
      <StyledBurger data-testid="burger-menu" open={open} onClick={handleOnclick}>
        <div />
        <div />
        <div />
      </StyledBurger>

      <MenuTray open={open} pathname={pathname} />
    </>

  )
};

export default BurgerMenu
