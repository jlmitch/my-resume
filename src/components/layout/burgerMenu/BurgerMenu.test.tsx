import React from 'react';
import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import BurgerMenu from './BurgerMenu';

jest.mock('./MenuTray', () => () => <div>Menu Tray</div>);

describe('BurgerMenu', () => {
  test('should display BurgerMenu content', () => {
    const { baseElement } = render(<BurgerMenu pathname="/cats/" />);

    expect(baseElement).toMatchSnapshot();
  });

  test('should match snapshot on click', () => {
    const { baseElement } = render(<BurgerMenu pathname="/cats/" />);

    userEvent.click(screen.getByTestId('burger-menu'));

    expect(baseElement).toMatchSnapshot();
  });
});
