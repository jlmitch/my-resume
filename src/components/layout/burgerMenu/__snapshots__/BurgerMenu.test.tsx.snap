// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BurgerMenu should display BurgerMenu content 1`] = `
<body>
  <div>
    <div
      class="sc-beqWaB fmVqgy"
      data-testid="burger-menu"
    >
      <div />
      <div />
      <div />
    </div>
    <div>
      Menu Tray
    </div>
  </div>
</body>
`;

exports[`BurgerMenu should match snapshot on click 1`] = `
<body>
  <div>
    <div
      class="sc-beqWaB fmVqgy"
      data-testid="burger-menu"
    >
      <div />
      <div />
      <div />
    </div>
    <div>
      Menu Tray
    </div>
  </div>
</body>
`;
